server/app/helpers/token_authentication_helper.rb

```diff
@@ -0,0 +1,46 @@
+# Validate access token in request headers
+#
+#
+require_relative '../routes/oauth2_api'
+require_relative '../services/logging'
+
+module TokenAuthenticationHelper
+
+  include OAuth2Api::Common
+  include Logging
+
+  def validate_access_token(*scopes)
+    # These only happen when in a "soft exclude" path where
+    # the headers are processed but request is not halted
+    # by the token authentication middleware.
+    
+    unless current_user
+      debug "No current user"
+      mime_halt(403, 'access_denied')
```

Comment:
It responds with correct mime type and format and adds [WWW-Authenticate header](http://self-issued.info/docs/draft-ietf-oauth-v2-bearer.html#authn-header).